### PR TITLE
Various GoReleaser artifact tweaks

### PR DESCRIPTION
**Pull request by matyasselmeci**
See #1271. This puts the contents of the pelican tarball into a pelican-<version>/ directory and also has the pelican apk/rpm/deb create and take ownership of the /etc/pelican directory.

### CODE REVIEW

**Comment by haoming29** [comment]
@matyasselmeci any thoughts for the comment above?

**Comment by matyasselmeci** [comment]
I can't find any comments. Was it on this PR?

**Comment by haoming29** [comment]
> I can't find any comments. Was it on this PR?

Shoot, my bad! I never hit the submit button

File: .goreleaser.yml [hunk 1]
```diff
@@ -106,6 +107,12 @@ nfpms:
             file_info:
               mode: 0644
             type: doc
+          - dst: "/etc/pelican"
```
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply .goreleaser.yml:110]: root permissions are required to install native packages in the first place. We can't really pre-create `~/.config/pelican` directories for people, though it is considered acceptable practice for a program to create those if they're missing (if we want that).

[the remark this answers is not in the document]
Comment (haoming29) [reply .goreleaser.yml:110]: Make sense. In this case, we should require user to have root access to install Pelican right? I'll need to update documentation for this.

[the remark this answers is not in the document]
Comment (matyasselmeci) [reply .goreleaser.yml:110]: Yes, you will need root to install pelican system-wide.

**Review by haoming29 — APPROVED** [review]
LGTM!

**Pull request merged**